src/game/gui/pilotpic.c

```diff
@@ -52,27 +43,45 @@ void pilotpic_select(component *c, int pic_id, int pilot_id) {
     int ret = sd_pic_load(&pics, filename);
     if(ret != SD_SUCCESS) {
         PERROR("Could not load PIC file %s: %s", filename, sd_get_error(ret));
-        return;
+        return ret;
     } else {
         DEBUG("PIC file %s loaded, selecting picture %d.", get_resource_name(pic_id), pilot_id);
     }
 
     // Create new
     const sd_pic_photo *photo = sd_pic_get(&pics, pilot_id);
+    sd_sprite_copy(sprite, photo->sprite);
+    palette_copy(pal, &photo->pal, 0, 48);
+    // Free pics
+    sd_pic_free(&pics);
+
+    return SD_SUCCESS;
+}
+
+void pilotpic_select(component *c, int pic_id, int pilot_id) {
+    pilotpic *local = widget_get_obj(c);
+
+    // Free old image
+    if(local->img != NULL) {
+        sprite_free(local->img);
+        omf_free(local->img);
+    }
+
     local->img = omf_calloc(1, sizeof(sprite));
-    sprite_create(local->img, photo->sprite, -1);
+    sd_sprite spr;
+    palette pal;
+    pilotpic_load(&spr, &pal, pic_id, pilot_id);
+
+    sprite_create(local->img, &spr, -1);
 
     // Position and size hints for the gui component
     // These are set on layout function call
     component_set_size_hints(c, local->img->data->w, local->img->data->h);
 
     // Save some information
     local->selected = pilot_id;
-    local->max = pics.photo_count;
+    local->max = 4; // TODO pics.photo_count;
```

Comment:
photo_count didn't work ?

Reply:
this whole widget needs a rewrite to keep the pic file in memory until its freed